routes/user.js

```diff
@@ -7,19 +7,19 @@ const {
   updateUser,
   deleteUser,
 } = require("../controllers/user");
-const { authenticate } = require("../middlewares/auth");
+const { authenticate, authorize } = require("../middlewares/auth");
 const {
   UserCreationValidation,
   UserUpdateValidation,
 } = require("../validations/user.validation");
 
 const userRouter = express.Router();
 
-userRouter.get("/", getAllUser);
-userRouter.get("/:id", getAUser);
+userRouter.get("/",authenticate, authorize, getAllUser);
```

Comment:
`authorize()` accepts an array of usertypes that can access the endpoint e.g `authorize([userTypes.Admin,userType.Student])`